### PR TITLE
Kotlin migration - part 2

### PR DESCRIPTION
**Pull request by matteosz**
Second part of the Kotlin migration. This PR migrates the rest of the files except for the \tests and \src\ui folders

### CODE REVIEW

**Review by jbsv — APPROVED** [review]
That's a lot of changes. I did what I could but couldn't finish it.
I just peeked one comment that is probably a forgotten one.

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/qrcode/PoPCHAQRCode.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    val uri = Uri.parse(data)
+    clientId = uri.getQueryParameter(FIELD_CLIENT_ID)!!
+    nonce = uri.getQueryParameter(FIELD_NONCE)!!
+    state = uri.getQueryParameter(FIELD_STATE)
+    responseMode = uri.getQueryParameter(FIELD_RESPONSE_MODE)
+    host = "${uri.host}${if (uri.port == -1) "" else ":${uri.port}"}"
+  }
+
+  override fun toString(): String {
+    return ("PoPCHAQRCode{" +
```
Comment (matteosz) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/model/qrcode/PoPCHAQRCode.kt:27]: Use string template here

**Review by matteosz — COMMENTED** [review]
Self-review

**Comment by jbsv** [comment]
Happy to approve as is, but can you add a separate PR to reduce the "23 new issues on FE2" ?

**Review by jbsv — APPROVED** [review]
Happy to merge if this code has been tested. Do you confirm that the FE2 app still runs in Android Studio?

**Review by pierluca — APPROVED** [review]
Couldn't review everything (for obvious reasons), but randomly sampled files look good 👍

**Pull request merged**